stackit/internal/services/objectstorage/credentialsgroup/resource.go

```diff
@@ -0,0 +1,335 @@
+	}
+
+	tflog.Info(ctx, "ObjectStorage credentials group deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id, credentials_group_id
+func (r *credentialsGroupResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing credentialsGroup",
+			fmt.Sprintf("Expected import identifier with format [project_id],[credentials_group_id], got %q", req.ID),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("credentials_group_id"), idParts[1])...)
+	tflog.Info(ctx, "ObjectStorage credentials group state imported")
+}
+
+func mapFields(credentialsGroupResp *objectstorage.CreateCredentialsGroupResponse, model *Model) error {
+	if credentialsGroupResp == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if credentialsGroupResp.CredentialsGroup == nil {
+		return fmt.Errorf("response credentialsGroup is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	credentialsGroup := credentialsGroupResp.CredentialsGroup
+
+	mapCredentialsGroup(credentialsGroup, model)
+	return nil
+}
+
+func mapCredentialsGroup(credentialsGroup *objectstorage.CredentialsGroup, model *Model) {
+	model.CredentialsGroupId = types.StringPointerValue(credentialsGroup.CredentialsGroupId)
+	model.URN = types.StringPointerValue(credentialsGroup.Urn)
+	model.Name = types.StringPointerValue(credentialsGroup.DisplayName)
+
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.CredentialsGroupId.ValueString(),
+	}
+	model.Id = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+}
+
+// enableProject enables object storage for the specified project. If the project already exists, nothing happens
+func (r *credentialsGroupResource) enableProject(ctx context.Context, model *Model) error {
+	projectId := model.ProjectId.ValueString()
+
+	// From the object storage OAS: Creation will also be successful if the project already exists, but will not create a duplicate
+	_, err := r.client.CreateProject(ctx, projectId).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create object storage project: %w", err)
+	}
+	return nil
+}
+
+// readCredentialsGroups gets all the existing credentials groups for the specified project,
+// finds the credentials group that is being read and updates the state. If the credentials group cannot be found, it throws an error
+func readCredentialsGroups(ctx context.Context, model *Model, projectId string, client *objectstorage.APIClient) error {
```

Comment:
I'd change the signature to
`func (r *credentialsGroupResource) readCredentialsGroups(ctx context.Context, model *Model) error`
so it matches other functions' signatures that need to use the client. Also, you can get `projectId` from the model, no need to have it as an input